doc/Language/rb-nutshell.pod

```diff
@@ -0,0 +1,944 @@
+      say "Yep!";
+      $x = 17         # No ; required before closing }
+    }                 # No ; required after closing } because of the newline
+    say "Done!";      # The ; is not required here if nothing follows
+
+=head3 Sigils
+
+In Ruby, variables use sigils primarily to indicate scope. C<$> for global
+scope, C<@@> for class scope, C<@> for instance scope, and no sigil for local
+variables (including parameters). The C<&> sigil is also used to indicate
+method references. Symbols are prefixed with C<:>, but they are not variable
+and so not really sigils.
+
+In Perl 6 sigils are primarily used to indicate a role that the contained value
+implements, indicating the type (or at least the interface) of the value. The
+sigils are invariant, no matter how the variable is being used - you can think
+of them as part of the variable's name.
+
+The scope of a variable is instead indicated by the declaration itself (C<my>,
+C<has>, C<our>, etc).
+
+=head4 C<$> Scalar
+
+The C<$> sigil is always used with "scalar" variables (e.g. C<$name>). These
+are single-value containers.
+
+This is the most general-purpose variable type, with no restrictions on its
+contents. Note that you can still address/use its contents, like C<$x[1]>,
+C<$x{"foo"}>, and C<$f("foo")>.
+
+=head4 C<@> Array
+
+The C<@> sigil is always used with "array" variables (e.g. C<@months>,
+C<@months[2]>, C<@months[2, 4]> for an array slice).
+
+=head4 C<%> Hash
+
+The C<%> sigil is always used with "hash" variables (e.g. C<%calories>,
+C<%calories<apple>>, C<%calories<pear plum>>).
+
+=head4 C<&> Sub
+
+TODO: This section isn't quite right
+
+The C<&> sigil is used very similarly to Ruby's C<&> to refer to the function
+object of a named subroutine/operator without invoking it, i.e. to use the name
+as a "noun" instead of a "verb":
+
+    foo_method = &foo;     # Ruby
+    my &foo_method = &foo; # Perl 6
+
+    callback: &say  # Ruby
+    callback => &say            # Perl 6 - & gives "noun" form of any sub
+
+Since Perl 6 does not allow adding/removing symbols in a lexical scope once
+it has finished compiling, there is no equivalent to Ruby's
+C<undef &foo;>, and the closest equivalent to Ruby's C<defined &foo>
+would be C<defined &::("foo")> (which uses the "dynamic symbol lookup"
+syntax). However, you can declare a mutable named subroutine with
+C<my &foo;> and then change its meaning at runtime by assigning to C<&foo>.
+
+In Ruby, the ampersand sigil can additionally be used to call subroutines
+in special ways with subtly different behavior compared to normal sub
+calls. In Perl 6 those special forms are no longer available:
+
+=head4 C<*> Slurpy params / argument expansion
+
+In Ruby you can declare a parameter to slurp the remainder of the passed
+arguments into an array using a C<*> prefix. It works the same way in Perl 6:
+
+    def foo(*args); puts "I got #{args.length} args!"; end # Ruby
+    sub foo(*@args) { say "I got #{@args.elems} args!" }   # Perl 6
```

Comment:
Is this right? Double-check. Also mention that slurping arguments separate positional and named params :(